evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -1443,6 +1425,7 @@ def test_automl_search_dictionary_oversampler(
         y_train=y,
         problem_type=problem_type,
         sampler_method="SMOTE Oversampler",
+        optimize_thresholds=False,
```

Comment:
Curious, what's going on here that all the optimize_thresholds need to be set to something off the default in all these tests that don't really have anything to do with the objective?

Reply:
@chukarsten Since we default to tuning the threshold, and this PR adds an objective to use for that even when the primary objective isn't tunable, we need to add this line in order to avoid tuning the threshold. When we tune, we call `predict_proba`, among other calls, and this fails since we're mocking `fit`, which results in a `pipeline not yet fitted` error. 

I chose to set `optimize_thresholds` to false rather than mocking `predict_proba` and the other methods that are involved in tuning.